### DISCUSSION TITLE
Request json is null

### DISCUSSION THREAD

**Discussion by josesalasdev** [encode/starlette#1610] [Ideas]
When calling this method `await request.json()` it is common to encounter this exception 
`raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)` when the body is null, my idea is to expose a flag that allows to control this error:

Path module: `starlette/requests.py`

```python
async def json(self, nullable: bool = False) -> typing.Any:
        if not hasattr(self, "_json"):
            body = await self.body()
            self._json = {} if (body == b'' and nullable) else json.loads(body)
        return self._json
```

this makes sense for you and the users ?
https://github.com/josesalasdev/starlette/commit/16c1c8e657ad0388e56d7c2349cf32918c4dd05a

**Comment by josesalasdev** [comment]
@tomchristie this makes sense for you?

**Comment by snosratiershad** [comment]
I think `{}` isn't a good response with `nullable` flag because it will returns same value for `b''` and `b'{}'` as request body.
Also `None` isn't good too, as it is the representation of `null`.
I think if we decide to add to add a flag like it and it's functionality, empty string would be a better response.
